src/gui/application.cpp

```diff
@@ -646,31 +549,106 @@ static void displayHelpText(const QString &t, std::ostream &stream = std::cout)
 #endif
 }
 
-void Application::showHelp()
+void Application::parseOptions(const QStringList &arguments)
 {
-    setHelp();
-    QString helpText;
-    QTextStream stream(&helpText);
-    stream << _theme->appName()
-           << QLatin1String(" version ")
-           << _theme->version() << endl;
+    QCommandLineParser parser;
+
+    QString descriptionText;
+    QTextStream descriptionTextStream(&descriptionText);
 
-    stream << QLatin1String("File synchronisation desktop utility.") << endl
-           << endl
-           << optionsC();
+    descriptionTextStream << _theme->appName() << QStringLiteral(" version ") << _theme->version() << endl;
+    descriptionTextStream << QStringLiteral("File synchronization desktop utility.");
 
     if (_theme->appName() == QLatin1String("ownCloud")) {
-        stream << endl
-               << endl
-               << "For more information, see http://www.owncloud.org";
+        descriptionTextStream << endl
+                              << endl
+                              << "For more information, see http://www.owncloud.org";
     }
 
-    displayHelpText(helpText);
-}
+    parser.setApplicationDescription(descriptionText);
+
+    auto helpOption = parser.addHelpOption();
+    auto versionOption = parser.addVersionOption();
+
+    // this little snippet saves a few lines below
+    auto addOption = [&parser](const QCommandLineOption &option) {
+        parser.addOption(option);
+        return option;
+    };
+
+    auto showSettingsOption = addOption({ { "s", "showsettings" }, "Show the settings dialog while starting." });
```

Comment:
We should translate the description.
Not sure weather we should translate the language options though :)

Reply:
The translations are loaded _after_ the options are parsed. There is no point in translating any of these. To make translations any useful, we'd have to implement some mechanism to reload the translations in case `--language` is set.

Alternatively, one could eliminate that parameter and replace it with an environment variable.

Reply:
ouch your right. 
Lets keep it untranslated for now.